### PR TITLE
Add 1D Burton mesh

### PR DESCRIPTION
**Pull request by cferenba**
This adds a basic 1D capability to the burton mesh specialization.  Has been tested successfully against an updated flecsale (which will be merged in a subsequent PR).

### CODE REVIEW

File: flecsi-sp/burton/burton_corner.h [hunk 1]
```diff
@@ -66,8 +67,153 @@ class burton_corner_t
   burton_corner_t( burton_corner_t && ) = delete;
   burton_corner_t & operator=( burton_corner_t && ) = delete;
 
+};
+
+////////////////////////////////////////////////////////////////////////////////
+//! \brief The burton_corner_t type provides an interface for managing and
```
Comment (cmsquared) [inline flecsi-sp/burton/burton_corner.h]: Extra "and"

File: flecsi-sp/burton/burton_types.h [hunk 2]
```diff
@@ -30,6 +30,265 @@ namespace burton {
...
+      cells_to_wedges = cells_to_corners,
+      vertices_to_wedges = vertices_to_corners,
+      edges_to_wedges = vertices_to_corners,
+      faces_to_wedges = vertices_to_corners,
+      wedges_to_cells = corners_to_cells,
+      wedges_to_vertices = corners_to_vertices,
+      wedges_to_edges = corners_to_vertices,
+      wedges_to_faces = corners_to_vertices,
+      // index spaces that are not used
+      vertices_to_edges = 7777,
```
Comment (cmsquared) [inline flecsi-sp/burton/burton_types.h]: What, no `Fnul`? :-)

Comment (charest) [reply flecsi-sp/burton/burton_types.h]: lol this is on me.  sorry I had to put something other than -1

Comment (cferenba) [reply flecsi-sp/burton/burton_types.h]: @cmsquared : Didn't you know that 7777 is the 16-bit version of `Fnul`, for use on reduced-precision GPUs and AI engines?  ;-)

File: flecsi-sp/burton/burton_corner.h (hunk 1 above)
Comment (cferenba) [reply flecsi-sp/burton/burton_corner.h]: Good catch.  I'll be doing another PR shortly that refactors this file, and I'll fix this then.

**Pull request merged**